src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,295 @@
+            private async ValueTask StartWritingByteValueInChunksAsync(ReadOnlyMemory<byte> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += chunkSize)
+                {
+                    int remainingBytes = Math.Min(chunkSize, value.Length - i);
+                    bool isFinalBlock = false ;
+
+                    // Take a chunk of bytes from the input value.
+                    ReadOnlyMemory<byte> chunk = value.Slice(i, remainingBytes);
+
+                    // Get unprocessed bytes from the buffer.
+                    ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = buffer;
+
+                    // If the buffer is not empty, then we copy the bytes from the buffer
+                    // to the current chunk being processed.
+                    if (!bytesNotProcessedFromPreviousChunk.IsEmpty)
+                    {
+                        // Create a new combined array to hold bytes from both previous and current chunks.
+                        byte[] combinedArray = new byte[bytesNotProcessedFromPreviousChunk.Length + chunk.Length];
+
+                        // Copy bytes from bytesNotProcessedFromPreviousChunk to the combined array
+                        bytesNotProcessedFromPreviousChunk.Span.CopyTo(combinedArray);
+
+                        // Copy bytes from chunk to the combined array, starting from the end of bytesNotProcessedFromPreviousChunk
+                        chunk.Span.CopyTo(combinedArray.AsSpan(bytesNotProcessedFromPreviousChunk.Length));
+
+                        WriteChunk(combinedArray, isFinalBlock);
+                    }
+                    else
+                    {
+                        WriteChunk(chunk.Span, isFinalBlock);
+                    }
+
+                    // Flush the writer if the buffer threshold is reached
+                    await this.jsonWriter.FlushIfBufferThresholdReachedAsync().ConfigureAwait(false);
+                }
+            }
+
+            /// <summary>
+            /// Writes a chunk of data, encoding it using Base64, and updates the buffer with any unprocessed bytes.
+            /// </summary>
+            /// <param name="chunk">The chunk of data to be encoded and written.</param>
+            /// <param name="isFinalBlock">A boolean indicating whether this is the final block of data.</param>
+            private void WriteChunk(ReadOnlySpan<byte> chunk, bool isFinalBlock)
+            {
+                // Encode the current chunk using Base64 and write it
+                this.jsonWriter.Base64EncodeAndWriteChunk(chunk, isFinalBlock, out int bytesNotWrittenFromPreviousChunk);
+
+                if (bytesNotWrittenFromPreviousChunk > 0)
+                {
+                    // Update the buffer with unprocessed bytes from the current chunk.
+                    buffer = chunk.Slice(chunk.Length - bytesNotWrittenFromPreviousChunk).ToArray();

```

Comment:
Would a `Span<byte>` be possible to use as a buffer in place of `byte[]` so we don't have to do .`ToArray()`?

Reply:
@gathogojr We can't use a `Span<byte>` as a field of a class. But we can use `ReadOnlyMemory<byte>`